compiler/rustc_infer/src/infer/outlives/obligations.rs

```diff
@@ -131,6 +132,25 @@ impl<'tcx> InferCtxt<'tcx> {
     ) -> Result<(), (E, SubregionOrigin<'tcx>)> {
         assert!(!self.in_snapshot(), "cannot process registered region obligations in a snapshot");
 
+        let normalized_caller_bounds: Vec<_> = outlives_env
+            .param_env
+            .caller_bounds()
+            .iter()
+            .filter_map(|clause| {
+                let bound_clause = clause.kind();
+                let ty::ClauseKind::TypeOutlives(outlives) = bound_clause.skip_binder() else {
```

Comment:
given that the caller bounds are literally only `TypeOutlives` - which is non-trivial/surprising, given that I expected it to contain `RegionOutlives` as well - , can you change `caller_bounds` in `TypeOutlives` to be `OutlivesPredicate<Ty<'tcx>, Region<'tcx>>` instead

Reply:
Yeah, the region outlives get put into the `FreeRegionMap` directly.